libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.Bot.Builder.Core.Extensions;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure
+{
+    /// <summary>
+    /// Models IStorage using Azure Storge Blobs
+    /// </summary>
+    /// <remarks>
+    /// The AzureBlobStorage implements State's IStorage using a single Azure Storage Blob Container.
+    /// Each entity or StoreItem is serialized into a JSON string and stored in an individual text blob.
+    /// Each blob is named after the StoreItem key which is encoded and ensure it conforms a valid blob name.
+    /// </remarks>
+    public class AzureBlobStorage : IStorage
+    {
+        private readonly static JsonSerializerSettings serializationSettings = new JsonSerializerSettings()
+        {
+            // we use all so that we get typed roundtrip out of storage, but we don't use validation because we don't know what types are valid
+            TypeNameHandling = TypeNameHandling.All
+        };
+
+        /// <summary>
+        /// The Azure Storage Blob Container where entities will be stored
+        /// </summary>
+        public Lazy<CloudBlobContainer> Container { get; private set; }

```

Comment:
I wouldn't expose this internal detail as a property. In fact, I'd suggest not using a `Lazy<>` here since there are `async` calls being made inside of it. More below.

Reply:
We found this being exposed in the TablStorage just for testability purpose. We'll take a look at a different approach for the same goal.